--- a/target/linux/qualcommax/dts/ipq8070-tl-er2260t.dts
+++ b/target/linux/qualcommax/dts/ipq8070-tl-er2260t.dts
@@ -226,26 +226,33 @@
 			phy_address = <3>;
 		};
 
-		sfp_1: port@4 {
+		port@4 {
 			port_id = <5>;
 			phy_address = <29>;
-			phy_i2c_address = <29>;
-			phy-i2c-mode;
-			media-type = "sfp";
 		};
 
-		sfp_2: port@5 {
+		port@5 {
 			port_id = <6>;
 			phy_address = <30>;
-			phy_i2c_address = <30>;
-			phy-i2c-mode;
-			media-type = "sfp";
 		};
 	};
 };
 
 &blsp1_i2c2 {
 	status = "okay";
+	clock-frequency = <100000>;
+
+	sfp1: sfp@50 {
+		compatible = "sff,sfp";
+		reg = <0x50>;
+		maximum-power-milliwatt = <3000>;
+	};
+
+	sfp2: sfp@51 {
+		compatible = "sff,sfp";
+		reg = <0x51>;
+		maximum-power-milliwatt = <3000>;
+	};
 };
 
 &edma {
@@ -286,7 +293,7 @@
 
 &dp5_syn {
 	status = "okay";
-	phy-handle = <&sfp_1>;
+	sfp = <&sfp1>;
 	label = "10ge1";
 	nvmem-cells = <&macaddr_wan>;
 	nvmem-cell-names = "mac-address";
@@ -294,7 +301,7 @@
 
 &dp6_syn {
 	status = "okay";
-	phy-handle = <&sfp_2>;
+	sfp = <&sfp2>;
 	label = "10ge2";
 	nvmem-cells = <&macaddr_lan>;
 	nvmem-cell-names = "mac-address";
